include/picongpu/particles/debyeLength/Check.x.cpp

```diff
@@ -0,0 +1,201 @@
+                auto estimate = estimateGlobalDebyeLength<T_Species>(cellDescription, minMacroparticlesPerSupercell);
+                if(!isPrinting)
+                    return;
+
+                auto const name = pmacc::traits::GetCTName_t<T_Species>::str();
+                log<picLog::PHYSICS>("Resolving Debye length for species \"%1%\"?") % name;
+
+                if(estimate.numUsedSupercells)
+                {
+                    auto const temperatureKeV = estimate.sumTemperatureKeV / estimate.sumWeighting;
+                    auto const debyeLength = estimate.sumDebyeLength / estimate.sumWeighting;
+                    auto maxCellSize = sim.pic.getCellSize()[0];
+                    // For 2D do not use grid size along z, as it is always resolved
+                    for(uint32_t d = 1; d < simDim; d++)
+                        maxCellSize = std::max(maxCellSize, sim.pic.getCellSize()[d]);
+                    auto const cellsPerDebyeLength = debyeLength / maxCellSize;
+                    auto const debyeLengthSI = debyeLength * sim.unit.length();
+                    log<picLog::PHYSICS>("Estimate used momentum variance in %1% supercells with at least %2% "
+                                         "macroparticles each")
+                        % estimate.numUsedSupercells % minMacroparticlesPerSupercell;
+                    auto const ratioFailingSupercells = 100.0f * static_cast<float>(estimate.numFailingSupercells)
+                        / static_cast<float>(estimate.numUsedSupercells);
+                    log<picLog::PHYSICS>("%1% (%2% %3%) supercells had local Debye length estimate not resolved"
+                                         " by a single cell")
+                        % estimate.numFailingSupercells % ratioFailingSupercells % "%";
+                    log<picLog::PHYSICS>("Estimated weighted average temperature %1% keV and corresponding "
+                                         "Debye length %2% m.\n"
+                                         "   The grid has %3% cells per average Debye length")
+                        % temperatureKeV % debyeLengthSI % cellsPerDebyeLength;
+                }
+                else
+                    log<picLog::PHYSICS>("Check skipped due to no supercells with at least %1% macroparticles")
+                        % minMacroparticlesPerSupercell;
+            }
+        };
+
+    } // namespace detail
+
+    /** Check Debye length resolution
+     *
+     * Compute and print the weighted average Debye length for the electron species.
+     * Print in how many supercells the locally estimated Debye length is not resolved with a single cell.
+     *
+     * The check is supposed to be called just after the particles are initialized at start of a simulation.
+     * The results are output to log<picLog::PHYSICS>.
+     *
+     * This function must be called from all MPI ranks.
+     *
+     * @param cellDescription mapping for kernels
+     */
```

Comment:
As a general discussion point, we have duplicated documentation here. This doesnt seem ideal

Reply:
No I forgot to remove the documentation in the cpp files. Let's merge #5090 and I will remove the documentation afterwards and check other cpp files too.